BST.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BST
+{
+    class BST
+    {
+
+        public Node root;

```

Comment:
Same encapsulation principle. Maybe it should be private and your BST behavior could be controller by exposed public methods?